Automattic/jetpack
# Update My Jetpack interstitial click events

Fixes linked issue. For more context, see: pbNhbs-9zN-p2#comment-20126

## Proposed changes:

* This PR updates `trackProductClick` and `trackBundleClick` to include the product slug of the current product if it's a paid product.

### Other information:


## Jetpack product discussion


pbNhbs-9zN-p2#comment-20126

## Does this pull request change what data or activity we track or use?


Kind of. It updates the call to a Tracks event to include one more piece of information

## Testing instructions:






* Spin up a JN site with Jetpack Beta pointing to this branch
* Visit /wp-admin/admin.php?page=my-jetpack#/add-search
* Open your browser's developer console on the Network tab, and filter by `t.gif`
* Click on "Get Jetpack Search" and confirm that the Tracks event has a `productSlug` property with the correct product slug (the same one used in the checkout screen)
* Go back to the Search interstitial, but now click on "Start for Free"
* Check the Network tab again. Now, the logged event contains an empty `productSlug`

<img width="773" alt="image" src="https://github.com/Automattic/jetpack/assets/5714212/9133734e-4b87-487b-a5ea-688fe12a644b">
<img width="792" alt="image" src="https://github.com/Automattic/jetpack/assets/5714212/07a5ce60-d110-463a-9e36-d285d0cde9ae">
